### PR TITLE
Make error statements and their formatting consistent

### PR DESCRIPTION
**Pull request by mancabizjak**
This commit re-formats error messages in `fmt.Errorf()` calls in order to conform to [code review recommendations for formatting error messages in Go](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

All the errors are now produced with `fmt.Errorf()` to achieve better consistency across the project. Calls to `errors.New()` were removed in favor of `fmt.Errorf()` calls.

I propose we follow the same convention (e.g. the use of `fmt.Errorf()` for definition of errors, and using short error messages starting with lower-cased letters and no punctuation at the end) in all new additions to the project. If you agree, we should "enforce" this during code review for all the subsequent PRs :slightly_smiling_face:

### CODE REVIEW

File: crypto/commitments/rsa_based.go [hunk 1]
```diff
@@ -43,7 +42,7 @@ func GenerateRSABasedQOneWay(nBitLength int) (func(*big.Int) *big.Int, func(*big
 		return nil, nil, nil, nil, err
 	}
 	if q.Cmp(rsa.N) < 1 {
-		return nil, nil, nil, nil, fmt.Errorf("Q must be > N")
+		return nil, nil, nil, nil, fmt.Errorf("q must be > n")
```
Comment (zitnik) [inline crypto/commitments/rsa_based.go]: I would rather have the names of mathematical variables in the same case as they are in comments. Which means upper-case `Q` and `N` here.

File: crypto/groups/qr_rsa.go [hunk 2]
```diff
@@ -36,7 +37,7 @@ type QRRSA struct {
 
 func NewQRRSA(P, Q *big.Int) (*QRRSA, error) {
 	if !P.ProbablyPrime(20) || !Q.ProbablyPrime(20) {
-		return nil, errors.New("P and Q must be primes")
+		return nil, fmt.Errorf("p and q must be primes")
```
Comment (zitnik) [inline crypto/groups/qr_rsa.go]: Same here.

File: crypto/commitments/rsa_based.go (hunk 1 above)
Comment (mancabizjak) [reply crypto/commitments/rsa_based.go]: Huh, you're right. Done!

File: crypto/groups/qr_rsa.go (hunk 2 above)
Comment (mancabizjak) [reply crypto/groups/qr_rsa.go]: Done!

**Pull request merged**